sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -360,3 +361,43 @@ async def negotiate(self):
         if returned_header == TLS_HEADER_FRAME:
             raise ValueError("Mismatching TLS header protocol. Excpected: {}, received: {}".format(
                 TLS_HEADER_FRAME, returned_header[1]))
+
+
+class WebSocketTransportAsync(WebSocketTransport):
+    async def _read(self, n, buffer=None, **kwargs): # pylint: disable=unused-arguments
+        """Read exactly n bytes from the peer."""
+        
+        length = 0
+        view = buffer or memoryview(bytearray(n))
+        nbytes = self._read_buffer.readinto(view)
+        length += nbytes
+        n -= nbytes
+        while n:
+            data = await asyncio.get_event_loop().run_in_executor(
+                None, self.ws.recv
```

Comment:
So this websocket library doesn't support async? Were there any that do?

Reply:
no - unfortunately, this doesn't.

There is no library which supports both sync and async. We have other libraries which are async only.

So the two choices we have are:
1. Go with the wrapper for sync implementation like here
2. Recommend different libraries for both sync and async.

Reply:
I'm leaning towards recommending different libraries...
We already do that for HTTP....
What's the quality/stability/maintenance of the async lib like?

Reply:
realized async library i was looking at doesn't support http proxy - so not an option for now 
https://github.com/aaugustin/websockets/issues/364

Reply:
Okay - can we open an issue to track this? At a minimum we should refactor this so that it will be compatible with an appropriate async implementation when it lands.

Reply:
Tracking it here https://github.com/Azure/azure-sdk-for-python/issues/24315